presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +122,33 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    public static Optional<EnumType> maybeGetEnumTypeFromLiteral(DereferenceExpression node, TypeManager typeManager)
+    {
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(node.getBase().toString()));
```

Comment:
`toString` is not safe here. `node.getBase()` potentially can be any expression. So you really want the input of this function to be just the `QualifiedName` and you want to use `QualifiedName.getPrefix()` here.